### ISSUE TITLE
Peer Review for Final Report- yx426

### ISSUE THREAD

**Issue by YihengXiao** [bakulcsingh/CervicalCancerBiopsyPrediction#14]
The project focuses on leveraging patients' medical history and behavioral features to predict the risk of cervical cancer. The team has analyzed 858 samples with 36 explanatory variables and tried to classified them into two categories whether or not should the patient be recommended to take biopsy test as the metric of cancer risk. Several classification models were utilized to generate satisfactory prediction. Moreover, the team also expressed plans for further study and potential commercial application. Overall, the project is attractive and meticulously conducted.

Things that are outstanding in the project:
1. Wonderful Visualization. In the first part, the team has shown 4 graphs to display the distribution of the input data, which paints clear panorama for readers. 

2. Comprehensive Models. The project is a classification problem. The team has chosen appropriate models, including perceptron, SVM (with weights) and random forest. Then, they fit the models in canerous and non-cancerous dataset. I embrace that that is a reasonable idea to deal with lopsided data. Plus, the final result shows a high accuracy.

3. Combating Overfitting. The report has introduced your method of preventing overfitting in details, such as class weights, dataset separation, using random forest and so forth. One of critical point of ORIE 4741 is to deal with overfitting. The team applied what we have learned in class to practical problems. 

Suggestions and possible improvements:
1. Model selection. The team selected preceptron as a tool of linear classification. However, the model is traditional with restrictions. SVM with various kernals not only obviously outperforms perceptron but reflects the priciple of it. Plus, logistic regression is also a standard classification model can be considered. So, I recommend that the team can replace perceptron with logistic.

2. Data cleaning. Firstly, The group stated in the report that they “ignore data points with "?”", which can be easily misunderstood. In the later part, the team "replaced the ‘?’ with values that would result in the least bias during classification and be consistent with the labels." That is confusing. Secondly, the team replaced them with the most commonly occurring entry for the
classification in the remaining data. I suggest that the team can simulate the distribution of the remaining data and fill in the missing data with random value drawn from that distribution.

3. Inprovements in extension part. Biopsy is highly correlated with the other three tests (Hinselmann test, Cytology and Schiller’s test). In reality, biopsy is at the verge of conducting operations that remove the tumor. It would be better to include less invasive and low correlated tests in the input dataset.

In a nutshell, the project is decently conducted and impressive.
